src/pip/_internal/resolution/resolvelib/candidates.py

```diff
@@ -376,7 +377,9 @@ def name(self) -> str:
 
     @property
     def version(self) -> CandidateVersion:
-        return self.dist.version
```

Comment:
Do we know why the `self.dist.version` line is slow? And how is it related to Python 3.11?

Reply:
I see, so the main difference is probably `pkg_resources` caches this value in-memory, while `importlib.metadata` needs to re-parse the file every time. The fix makes sense then.

It would be best to modify the news fragment to reflect that this is about `importlib.metadata` (and not 3.11 per se), and when pip repeatedly accesses an already-installed distribution (and not all “large environments” are affected, only if that environment has a lot of overlaps with the to-be-installed requirement set).

Reply:
Done (and rebased on main).